Sources/App/Views/ReadyForSwift6/ReadyForSwift6Show+View.swift

```diff
@@ -34,14 +35,88 @@ extension ReadyForSwift6Show {
+                .p("This chart shows the total number of packages that will compile with  Swift 6:"),
+                .readyForSwift6Chart(chartIdentifier: "readyforswift6-compatiblepackages"),
+                .h3("Total Swift 6 concurrency errors"),
+                .p("This chart shows the total number of Swift concurrency errors across the entire selection of testing packages:"),
+                .readyForSwift6Chart(chartIdentifier: "readyforswift6-totalerrors"),
+                .h3("List of compatible packages"),
+                .p("Here are all the compatible packages!")
+            )
+        }
+    }
+}
+
+private extension Node where Context: HTML.BodyContext {
+    static func readyForSwift6Chart(chartIdentifier: String) -> Self {
+        let specPath = Current.fileManager.workingDirectory().appending("Resources/Charts/\(chartIdentifier)-spec.json")
+        let dataPath = Current.fileManager.workingDirectory().appending("Resources/Charts/\(chartIdentifier)-data.json")
+        guard let chartSpec = Current.fileManager.contents(atPath: specPath)?.compactJson(),
+              let chartData = Current.fileManager.contents(atPath: dataPath)?.compactJson()
+        else {
+            return .p(
+                .text("Failed to load "),
+                .code(.text(chartIdentifier)),
+                .text(".")
             )
         }
+
+        let script = """
+        var spec = JSON.parse(document.getElementById('vega-spec-\(chartIdentifier)').textContent)
+        spec.data[0].values = JSON.parse(document.getElementById('vega-data-\(chartIdentifier)').textContent)
+        new vega.View(vega.parse(spec), { renderer: 'canvas' }).initialize('#vega-chart-\(chartIdentifier)').run()
+        """
+
+        return .group(
+            .div(
+                .class("vega-chart"),
+                .id("vega-chart-\(chartIdentifier)")
+            ),
+            .script(
+                .id("vega-spec-\(chartIdentifier)"),
+                .attribute(named: "type", value: "application/json"),
+                .raw(chartSpec)
+            ),
+            .script(
+                .id("vega-data-\(chartIdentifier)"),
+                .attribute(named: "type", value: "application/json"),
+                .raw(chartData)
+            ),
+            .script(
+                .raw(script.compactJavaScript())
+            )
+        )
+    }
+}
+
+private extension String {
+    func compactJavaScript() -> String {
+        self.split(separator: "\n")
+            .map { $0.trimmingCharacters(in: .whitespaces) }
+            .joined(separator: ";")
+    }
+}
+
+private extension Data {
+    func compactJson() -> String? {
+        guard let json = try? JSONSerialization.jsonObject(with: self),
+              let compactedJsonData = try? JSONSerialization.data(withJSONObject: json),
+              let compactJson = String(data: compactedJsonData, encoding: .utf8)
+        else { return nil }
+        return compactJson
```

Comment:
What's this doing exactly? Looks to me like it's taking JSON loaded from disk and essentially passing it through to a String? Couldn't we just pass the String from the file?